### ISSUE TITLE
dbt v1.0.0

### ISSUE THREAD

**Issue by jdub** [sqlfluff/sqlfluff#2048]
Sorry, very boring enhancement request… [dbt v1.0.0](https://github.com/dbt-labs/dbt-core/releases/tag/v1.0.0) is out, but sqlfluff-templater-dbt v0.8.2 is not compatible with it.

```
ImportError: cannot import name 'PROFILES_DIR' from 'dbt.config.profile'
```

**Comment by jpy-git** [comment]
@jdub given that dbt 1.0.0 is a new major release, the changes to it's API and hence the required changes to sqlfluff-templater-dbt are non-trivial.

To set expectations, we're targeting a new release in the next week and we are likely to pin the dbt dependency at `<1.0.0` for that release whilst we work through the necessary updates to ensure compatibility. But the new dbt version is on the maintainers radar and something we're looking in to 😄

**Comment by jdub** [comment]
Yay, thank you!

**Comment by jpy-git** [comment]
@alanmcruickshank this is the issue for introducing dbt 1.0.0 compatibility to SQLFluff. I've closed #2052 so you can handle this, but there was some initial discussion in there that may be helpful (e.g. adding the relevant tests).  Thanks! 😄
